song_rater/views.py

```diff
@@ -15,3 +16,24 @@ def index(request):
 class SongListView(ListView):
     model = Song
     context_object_name = 'song_list'
+
+
+RatingForm = modelform_factory(Rating, exclude=["song"])
+
+
+def song_detail(request, id):
+    if request.method == "POST":
+        form = RatingForm(request.POST)
+        if form.is_valid():
+            rating = form.save(commit=False)
+            rating.song_id = id
+            rating.save()
+            return redirect("song_rater:song_list")
+    else:
+        form = RatingForm()
+
+    context = {"details": get_object_or_404(Song, pk=id),
+               "form": form,
+               }
+
+    return render(request, 'song_rater/song_detail.html', context)
```

Comment:
OPTIONAL: Have you looked if you can replace this with a GenricView by Django? Maybe `DetailView` would be a fit.  

I think it might lead to cleaner code, but I think you will touch this code again in follow-up Pull Requests. It might be best to keep this as-is and maybe try to refactor it at the end. :)

Reply:
As discussed in a call @WangRemy  won't look into this.